p2p/src/sync/tests/peer_events.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    error::PeerError, net::default_backend::types::PeerId, sync::tests::helpers::SyncManagerHandle,
+    P2pError,
+};
+
+// Check that the header list request is sent to a newly connected peer.
+#[tokio::test]
+async fn connect_peer() {
+    let mut handle = SyncManagerHandle::start().await;
+
+    let peer = PeerId::new();
+    handle.connect_peer(peer).await;
+}
```

Comment:
Maybe it's best to assert the state after doing these operations, like assert that the peer exists

Reply:
There is an assertion inside the `connect_peer` function:
```rust
pub async fn connect_peer(&mut self, peer: PeerId) {
    self.try_connect_peer(peer);

    let (sent_to, message) = self.message().await;
    assert_eq!(peer, sent_to);
    assert!(matches!(message, SyncMessage::HeaderListRequest(_)));
}
```
Also I have added more assertions everywhere.

Reply:
Keep in mind that we can't rely on in-code asserts for tests. This is an anti-pattern. The idea is that code changes, but tests should be the judge on whether the code is valid.

If you've depended on this in other pleces, please track that and fix it.

Reply:
> Keep in mind that we can't rely on in-code asserts for tests.

Sure, but it is actually a part of the tests. The actual function is called `register_peer` and `connect_peer` is just a helper for the tests.